---
# The contents of this file cannot be in user_check.yml because of this issue
# https://github.com/ansible/ansible/issues/19391#issuecomment-282196876
- name: Apache VirtualHost checks
  block:

    - name: "Debug the VirtualServer's for {{ user.key }}"
      ansible.builtin.debug:
        msg: "VirtualServer: {{ vhost | to_nice_yaml }}"
        verbosity: 2
      loop: "{{ user.value.users_apache_virtual_hosts | dict2items }}"
      loop_control:
        loop_var: vhost
        label: "{{ vhost.key }}"

    - name: "Check users_apache_virtual_hosts variables using meta/apache_virtual_host_argument_specs.yml for {{ user.key }}"
      ansible.builtin.validate_argument_spec:
        argument_spec: "{{ lookup('ansible.builtin.file', 'meta/apache_virtual_host_argument_specs.yml') | from_yaml }}"
        provided_arguments: "{{ vhost.value }}"
      loop: "{{ user.value.users_apache_virtual_hosts | dict2items }}"
      loop_control:
        loop_var: vhost
        label: "{{ vhost.key }}"

    - name: "Loop thought the VirtualHosts for {{ user.key }}"
      ansible.builtin.include_tasks: user_check_vhost.yml
      loop: "{{ user.value.users_apache_virtual_hosts | dict2items }}"
      loop_control:
        loop_var: vhost
        label: "{{ vhost.key }}"

  # rescue:

  #   - name: There was a failure validating the arg spec
  #     ansible.builtin.debug:
  #       msg: "The users role meta/apache_virtual_host_argument_specs.yml vaildating failed for the user {{ user.key }}"

  tags:
    - users
...
